tools/site_admin/copy_pages.php

```diff
@@ -282,12 +282,13 @@ function do_stuff( $projectid_, $from_image_, $page_name_handling,
 
         // ----------------------
 
-        $res = mysqli_query(DPDatabase::get_connection(), sprintf("
+        $sql = sprintf("
             SELECT image, fileid
             FROM %s
```

Comment:
I would embed the `$projectid` directly in the string and drop the `sprintf()`. That way the sprintf is reserved for escaping values and not the table or column names. https://www.pgdp.net/w/index.php?title=DP_Code_Best_Practices#Column_names_as_variables

Reply:
Ditto other places where we build SQL in this file.

Reply:
Done.